src/App.tsx

```diff
@@ -15,71 +15,85 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export class App extends React.Component {
+  state = {
+    selectedGood: 'Jam',
+  }
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  handleClick = (good: string) => {
+    return () => this.setState({ selectedGood: good });
+  }
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleClear = () => {
+    this.setState({
+      selectedGood: '',
+    });
+  }
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+    return (
+      <main className="section container">
+        {this.state.selectedGood
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              {`${this.state.selectedGood} is selected`}
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                onClick={this.handleClear}
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+              />
+            </h1>
+          )
+          : <h1 className="title">No goods selected</h1>}
+        <table className="table">
+          <tbody>
+            {goods.map((good) => {
+              return (
+                <tr
+                  key={good}
+                  data-cy="Good"
+                  className={good === selectedGood
+                    ? 'has-background-success-light'
+                    : undefined}
+                >
+                  {good === this.state.selectedGood
```

Comment:
Since there are no button components for different scenarios,  I'd rather modify a button properties instead of rendering different buttons. 
```
                        <button
                          onClick={good === this.state.selectedGood ? this.handleClear : ....}
                          data-cy={good === this.state.selectedGood ? ....}
                          type="button"
                          className=`button ${good === this.state.selectedGood && ....}`
                        >
                          {good === this.state.selectedGood ? ...}
                        </button>
```
while this approach may be arguable, there is definitely no need to render `<td>` twice, you can place your render condition inside.